### PR TITLE
added async fetches via redux

### PR DESCRIPTION
**Pull request by borishere**

### CODE REVIEW

File: src/Components/Filter/Filter.tsx [hunk 1]
```diff
@@ -9,23 +11,36 @@ interface Props {
...
       return genre;
     });
 
     setGenres([...updatedGenres]);
   };
 
+  useEffect(() => {
+    dispatch(
+      setFilter(
+        genres
```
Comment (MibenCoop) [inline src/Components/Filter/Filter.tsx]: I would separate this into a separate variable before the dispatch to reduce chain

File: src/Components/Header/Header.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import './style.scss';
 
 interface Props {
   onValueChange?: (val: Imovie) => void;
-  selectedMovie: Imovie | null;
+  selectedMovie: Imovie | undefined;
```
Comment (MibenCoop) [inline src/Components/Header/Header.tsx]: I think, it could be just optional property

File: src/Components/Filter/Filter.tsx (hunk 1 above)
Comment (borishere) [reply src/Components/Filter/Filter.tsx]: Thanks for the tip.

File: src/Components/Header/Header.tsx (hunk 2 above)
Comment (borishere) [reply src/Components/Header/Header.tsx]: Thanks for the tip.

**Pull request merged**